### PR TITLE
Project setup

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR adds deployment-related code moved from `core-java` repository.

### CODE REVIEW

**Comment by alexander-yevsyukov** [comment]
@armiol, please add configs for CI environment, as we discussed.

File: .gitignore [hunk 1]
```diff
@@ -1,22 +1,42 @@
...
+!/.idea/codeStyleSettings.xml
 
-# BlueJ files
-*.ctxt
+# Gradle interim configs
+.gradle/
 
-# Mobile Tools for Java (J2ME)
-.mtj.tmp/
+# Maven deployment build directory
```
Comment (armiol) [inline .gitignore]: Let's remove this section.

File: .gitignore [hunk 2]
```diff
@@ -1,22 +1,42 @@
...
 
-# Package Files #
-*.jar
-*.war
-*.ear
-*.zip
-*.tar.gz
-*.rar
+# Generated source code
+*/generated/*
```
Comment (armiol) [inline .gitignore]: This should be `generated/`.

File: .gitignore [hunk 3]
```diff
@@ -1,22 +1,42 @@
...
 
-# virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
-hs_err_pid*
+# Gradle build files
+build/*
```
Comment (armiol) [inline .gitignore]: This should be `build/` and the next line may be removed at all.

File: .gitignore [hunk 4]
```diff
@@ -1,22 +1,42 @@
...
+*/build/*
+
+# Class files produced by the IDE
+classes/*
```
Comment (armiol) [inline .gitignore]: This should be `classes/`.

**Pull request merged**